### ISSUE TITLE
Hybrid approach

### ISSUE THREAD

**Issue by dyorgio** [caprica/vlcj#347]
Hello,

This is just a question, not a real issue :).

Can we use vlcj player in native mode (without frames data interception, like swing player) for 
fullscreen fast playback, but when user pause video (for exemple)  get a "screenshot" of movie 
to draw swing components over it?

Is it possible to mix native x embeded implementations?

**Comment by caprica** [comment]
You should look at the overlay component provided by the embedded media player.

This keeps a transparent window position on top of the video surface, and you can draw into it.

There are limitations. Check the github sources for OverlayTest and read the Javadoc and the source.

**Comment by dyorgio** [comment]
Hi caprica :),

The idea is be fast with native player and use de power of embedded (overlays, manipulation of output)
when user pause the video.

I'm thinking about user this approuch on raspberry-pi (low cpu power, but vlc now supports 1080p gpu accelerated playback).

What do you think? there is a way to do it?

I don't want to take a native SO's screenshot :P.

**Comment by caprica** [comment]
I still don't understand, sorry...

In vlcj terms, "native" = "embedded".

Using a "direct" media player is *one* way to manipulate the output and draw overlays - but this is neither "native" nor "embedded".

If you want to draw overlays on a "native" media player, you can use the overlay method I described in my previous comment. To be clear, this particular overlay approach is designed for vlcj "embedded" media players.

So in short, I would use an embedded media player with a heavyweight overlay. I don't see any need for a hybrid approach.

**Comment by dyorgio** [comment]
Ok, bad terms...sorry.

My idea is never use draw on java surfaces (Canvas, Component, etc...) because it is very slow on raspberry. (<2 fps)

So, I probably will use a player that not supports manipulation of frames.

My idea is mix this with a way to get the last frame when user pauses the video, than I could transform it in a 3D env and overlay with translucid controls.

**Comment by caprica** [comment]
OK, but I really don't see what you expect vlcj to do in this scenario.

**Comment by dyorgio** [comment]
I believe that player.saveSnapshot() will be a good way to solve :P.

**Comment by caprica** [comment]
player.getSnapshot() will return you an actual BufferedImage rather than a File if that's of more use.
